services/admissioncontrol/admission_policy/admission_control_policy.go

```diff
@@ -0,0 +1,86 @@
+package admission_policy
+
+import (
+	"fmt"
+	"net/http"
+)
+
+const (
+	admissionControlPolicyResourcePath = "kubernetes/admissionControl/policy"
+)
+
+type AdmissionControlPolicyRequest struct {
+	TargetId        string   `json:"targetId"`
+	TargetType      string   `json:"targetType,omitempty"`
+	RulesetId       int      `json:"rulesetId"`
+	NotificationIds []string `json:"notificationIds"`
+	Action          string   `json:"action"`
+	RulesetPlatform string   `json:"rulesetPlatform"`
+}
+
+type AdmissionControlPolicyResponse struct {
+	ID              string   `json:"id"`
+	TargetId        string   `json:"targetId"`
+	TargetType      string   `json:"targetType"`
+	RulesetId       int      `json:"rulesetId"`
+	Action          string   `json:"action"`
+	NotificationIds []string `json:"notificationIds"`
+	ErrorMessage    string   `json:"errorMessage"`
+}
+
+func (service *Service) GetAdmissionControlPolicy(id string) (*AdmissionControlPolicyResponse, *http.Response, error) {
+	v := new(AdmissionControlPolicyResponse)
+	path := fmt.Sprintf("%s/%s", admissionControlPolicyResourcePath, id)
+	resp, err := service.Client.NewRequestDo("GET", path, nil, nil, v)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	return v, resp, nil
+}
+
+func (service *Service) GetAllAdmissionControlPolicies() (*[]AdmissionControlPolicyResponse, *http.Response, error) {
+	v := new([]AdmissionControlPolicyResponse)
+	resp, err := service.Client.NewRequestDo("GET", admissionControlPolicyResourcePath, nil, nil, v)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	return v, resp, nil
+}
+
+func (service *Service) Create(body *AdmissionControlPolicyRequest) (*AdmissionControlPolicyResponse, *http.Response, error) {
+	v := new([]AdmissionControlPolicyResponse)
+	resp, err := service.Client.NewRequestDo("POST", admissionControlPolicyResourcePath, nil, []*AdmissionControlPolicyRequest{body}, v)
```

Comment:
as i understand, our endpoint api for policy should get list of AdmissionControlPolicyRequestModel as shown here : https://github.com/dome9/dome9_backend/blob/master/Dome9.WebApi/K8s/AdmissionControl/AdmissionControlController.cs#L102, if we just sent simply body, maybe we will get an error from our api that we don't match the required api endpoint, so there is a need to wrap it as array of AdmissionControlPolicyRequest struct with one element (body)

Reply:
are you sure? body is already of type AdmissionControlPolicyRequest so it should have the right fields. looking at one example I don't see it done like this https://github.com/dome9/dome9-sdk-go/blob/master/services/rulebundles/rule_bundles.go#L84

Reply:
i can try , but think it will make some error, also you can see this example here, very similar to our ac policy : 
https://github.com/dome9/dome9-sdk-go/blob/master/services/compliance/continuous_compliance_policy/continuous_compliance_policy.go#L52